nni/nas/experiment/config/experiment_config.py

```diff
@@ -42,34 +52,51 @@ def __init__(self, training_service_platform: Union[str, None] = None,
                  execution_engine: Union[str, ExecutionEngineConfig] = 'py',
                  **kwargs):
         super().__init__(training_service_platform, **kwargs)
-        self.execution_engine = execution_engine
+
+        if not utils.is_missing(self.execution_engine):
+            # this branch means kwargs is not {} and self.execution_engine has been assigned in super(),
+            # reassign it because super() may instantiate ExecutionEngineConfig by mistake
+            self.execution_engine = init_execution_engine_config(kwargs['executionEngine'])
+            del kwargs['executionEngine']
+        elif isinstance(execution_engine, str):
+            self.execution_engine = init_execution_engine_config(execution_engine)
+        else:
+            self.execution_engine = execution_engine
+
+        self._is_complete_config = False
```

Comment:
What does `_is_complete_config` mean?

Reply:
comment already added, it is only for resume and view, as in resume and view mode config is already canonicalized (currently canonicalize is not idempotent for proper validation, leave it to the refactor of exp config)